### PR TITLE
[pytorch/caffe2] Update extensions for v1.6.0-rc1

### PR DESCRIPTION
**Pull request by kparichay**
Update pytorch extension for pytorch v1.6.0-rc1
Also pytorch and caffe2 headers do not follow redundant declarations
So, remove redundant declaration warning flag when compiling with them

Update caffe2 extension to remove warning based on api update for version 1.6.0-rc1

Related issue : #2411

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/meson.build [hunk 1]
```diff
@@ -140,6 +140,20 @@ if pytorch_support_is_available
 
   nnstreamer_filter_torch_deps = pytorch_support_deps + [glib_dep, gst_dep, nnstreamer_dep]
 
+  pytorch_compile_args = []
+
+  pg_pkgconfig = find_program('pkg-config')
+  cmd = run_command(pg_pkgconfig, ['--print-errors', '--exists', 'pytorch >= 1.2.0'])
+  if cmd.returncode() == 0
+    pytorch_compile_args += '-DPYTORCH_VER_ATLEAST_1_2_0=1'
+  endif
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/meson.build:149]: Can you check pytorch version directly with this: https://mesonbuild.com/Dependencies.html#dependencies-using-config-tools ??

File: meson.build [hunk 2]
```diff
@@ -288,6 +250,49 @@ foreach feature_name, data : features
 
 endforeach
 
+# Define warning flags for c and cpp
```
Comment (myungjoo) [inline meson.build:253]: Why do you need to move this block?

File: ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc [hunk 3]
```diff
@@ -477,12 +482,20 @@ TorchCore::invoke (const GstTensorMemory * input, GstTensorMemory * output)
       ml_loge ("Output Tensor Information is not valid");
       return -2;
     }
+#ifdef PYTORCH_VER_ATLEAST_1_2_0
+  } else if (output_value.isList ()) {
+    c10::ArrayRef < torch::jit::IValue > output_ref_list =
+      output_value.toListRef ();
+    std::vector < torch::jit::IValue > output_list (
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc:489]: I found that the TAOS-CI have not checked the clang format '.cc' files (https://github.com/nnstreamer/TAOS-CI/pull/632).
Anyway, you have to fix the format later.

File: ext/nnstreamer/tensor_filter/meson.build (hunk 1 above)
Comment (kparichay) [reply ext/nnstreamer/tensor_filter/meson.build:149]: Updated.

File: meson.build (hunk 2 above)
Comment (kparichay) [reply meson.build]: pytorch and caffe2 do not comply with `-wredundant-declarations`. The patch tries to add it when pytorch and caffe2 extensions aren't compiled. Could not find a way to remove already added project arguments.
Move back the default declared flags to the original location.

File: ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc (hunk 3 above)
Comment (kparichay) [reply ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc]: Applying the clang format syntax to the file produces changes beyond the scope of this PR.
I have just applied clang-format to this changes introduced in this PR. 
Extensions will have to updated with syntax in a separate PR.

Comment (dongju-chae) [reply ext/nnstreamer/tensor_filter/tensor_filter_pytorch.cc]: Sure, I didn't mean that you should change the format in this PR.

**Pull request merged**